docs/rule/attributes-order.md

```diff
@@ -0,0 +1,60 @@
+# attributes-order
+
+Element attributes must be ordered, by default the order is:
+
+For angle bracket invocation syntax:
+
+- arguments
+- attributes
+- element modifiers
+- ...attributes
+- ?attributes
+
+For mustache/curly invoication syntax:
+
+- attributes
+
+## Configuration
+
+The following values are valid configuration:
+
+- attributeOrder: default [`arguments`, `attributes`, `modifiers`, `splattributes`]
+  - [`attributes`] for curly invocation
+- alphabetize: default `true` - `true` to enforce alphabetically ordered values within attributes
+
+## Examples
+
+This rule **forbids** the following:
+
+```hbs
+<MyComponent ...attributes @name='Hello' />
```

Comment:
The position of `...attributes` is significant in that the other arguments will override or be overridden by anything in `...attributes` depending on whether it comes before or after, right? Is that the only type of attribute whose ordering is meaningful? If reordering any types of attributes can cause a behavior change, we should call that out in the rule doc.

Reply:
One follow-up to this: assuming this is true, then there are two concerns I think we need to address:

* Should we just allow the spread to be in any position so the user can achieve their desired behavior? I.e. don't enforce spread ordering?
* An autofix shouldn't change the code behavior as that could cause breakages, so we need to make sure we don't move anything from one of the spread to the other.

Reply:
Autofix has been implemented in a way that the spread is left in the first position or last position while other attribute types are ordered per configuration.  When not in the first or last position, checking does not occur as ordering becomes exponentially more complicated.  Such ordering should be extremely rare in any case.

Reply:
Thanks! As long as spread can be in any position, I'm happy. Can you update these examples? It seems these are outdated now.